src/mlpack/core/data/normalize_labels_impl.hpp

```diff
@@ -39,32 +40,32 @@ void NormalizeLabels(const RowType& labelsIn,
   // we'll resize it back down to its actual size.
   mapping.set_size(labelsIn.n_elem);
   labels.set_size(labelsIn.n_elem);
+  // Map for mapping labelIn to their label
+  std::map<eT, size_t> hastTable;
   size_t curLabel = 0;
   for (size_t i = 0; i < labelsIn.n_elem; ++i)
   {
-    bool found = false;
-    for (size_t j = 0; j < curLabel; ++j)
+    // If labelsIn[i] aldeardy there in Map then just its label
+    if (hastTable[labelsIn[i]] != 0)
     {
-      // Is the label already in the list of labels we have seen?
-      if (labelsIn[i] == mapping[j])
-      {
-        labels[i] = j;
-        found = true;
-        break;
-      }
+      labels[i] = hastTable[labelsIn[i]]-1;
     }
-
-    // Do we need to add this new label?
-    if (!found)
+    else
     {
-      mapping[curLabel] = labelsIn[i];
+      // If labelsIn[i] not there then add it to Map
+      hastTable[labelsIn[i]] = curLabel+1;
       labels[i] = curLabel;
       ++curLabel;
     }
   }
-
   // Resize mapping back down to necessary size.
   mapping.resize(curLabel);
+  // Mapping array created with encoded labels
+  for (auto it=hastTable.begin(); it != hastTable.end(); ++it)
```

Comment:
Can you fix the style here please with the spaces?  It should read `auto it = hastTable.begin()`.  That same horizontal whitespace bit applies a few other places too.

Reply:
Fixed all style issue,PTAL